### PR TITLE
Implement explicit enums

### PR DESCRIPTION
**Pull request by andreibancioiu**
Implement explicit enums e.g.:

```
"OperationCompletionStatus": {
            "type": "explicit-enum",
            "variants": [
                {
                    "docs": [
                        "indicates that operation was completed"
                    ],
                    "name": "completed"
                },
                {
                    "docs": [
                        "indicates that operation was interrupted prematurely, due to low gas"
                    ],
                    "name": "interrupted"
                }
            ]
        },
```

References:
 - https://github.com/multiversx/mx-sdk-js-core/issues/519
 - https://github.com/multiversx/mx-sdk-rs/pull/1113

### CODE REVIEW

File: multiversx_sdk/abi/enum_value.py [hunk 1]
```diff
@@ -108,3 +108,8 @@ def __iter__(self):
 
         for field in self.fields:
             yield (field.name, field.value)
+
+
+class _EnumPayload(SimpleNamespace):
+    def __int__(self):
```
Comment (andreibancioiu) [inline multiversx_sdk/abi/enum_value.py:114]: This way, people can convert an `EnumValue` to an `int` (quite useful).

File: multiversx_sdk/abi/explicit_enum_value.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+from multiversx_sdk.abi.string_value import StringValue
+
+
+class ExplicitEnumValue(StringValue):
+    def __init__(self, value: str = "") -> None:
+        self.value = value
+
+    def __eq__(self, other: Any) -> bool:
+        return isinstance(other, ExplicitEnumValue) and self.value == other.value
```
Comment (andreibancioiu) [inline multiversx_sdk/abi/explicit_enum_value.py:11]: Currently, we don't differentiate between different types of enums if they have the same value. We'll brainstorm this a bit (could be perfectly fine in practice).

File: multiversx_sdk/abi/__init__.py [hunk 3]
```diff
@@ -6,8 +6,8 @@
 from multiversx_sdk.abi.biguint_value import BigUIntValue
 from multiversx_sdk.abi.bool_value import BoolValue
 from multiversx_sdk.abi.bytes_value import BytesValue
-from multiversx_sdk.abi.code_metadata_value import CodeMetadataValue
```
Comment (popenta) [inline multiversx_sdk/abi/__init__.py:9]: why was this deleted?

**Review by popenta — APPROVED** [review]

File: multiversx_sdk/abi/__init__.py (hunk 3 above)
Comment (andreibancioiu) [reply multiversx_sdk/abi/__init__.py:9]: By mistake :see_no_evil: 

Fixed.

**Review by popenta — APPROVED** [review]

**Review by danielailie — APPROVED** [review]
